### ISSUE TITLE
DelegatedMethod0 is not serializable

### ISSUE THREAD

**Issue by yujiantao** [hbutani/icebergSQL#5]
This project is very helpful for us to deploy iceberg with spark sql support.
When we test `join` with icebergSQL,  we found serialization issue as below. After we make DelegatedMethod0 extend Serializable, the problem is still there, but this time the unserializable method is java.lang.reflect.Method, which is unchangeable. Is there any way to workaround?
```
select * from store_sales_out a join store_sales b on a.ss_item_sk=b.ss_item_sk where a.ss_item_sk=17908
```
Caused by: java.io.NotSerializableException: org.apache.spark.sql.iceberg.utils.DelegationUtils$DelegatedMethod0
Serialization stack:
- object not serializable (class: org.apache.spark.sql.iceberg.utils.DelegationUtils$DelegatedMethod0, value: org.apache.spark.sql.iceberg.utils.DelegationUtils$DelegatedMethod0@692ff367)

##After we make DelegatedMethod0 extend Serializable
Caused by: java.io.NotSerializableException: java.lang.reflect.Method
Serialization stack:
	- object not serializable (class: java.lang.reflect.Method, value: public abstract org.apache.spark.rdd.RDD org.apache.spark.sql.execution.SparkPlan.doExecute())
	- field (class: org.apache.spark.sql.iceberg.utils.DelegationUtils$DelegatedMethod0, name: method, type: class java.lang.reflect.Method)

**Comment by hbutani** [comment]
Oh good catch; busy today, tomorrow; will take a look over the weekend.

**Comment by hbutani** [comment]
Fixed in commit 63c26ae00bb03bb7d05bfbeac4fe6787b019d505
